### PR TITLE
Added menu in productListsActivity to import csv file containing list

### LINKED ISSUE #2294: Add the ability to reimport a List CSV export into the App
Add the ability to reimport a List CSV export into the App
Loop for the barcode column, and re-add the items to a list.
A simple way to exchange a list between 2 people (for shopping)

### PR DESCRIPTION
**Pull request by prashantkh19**
## Related issues and discussion
fixes #2294

### CODE REVIEW

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/ProductListsActivity.java [hunk 1]
```diff
@@ -145,4 +164,99 @@ public void onRightClicked(int position) {
...
+            selectCSVFile();
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    private void selectCSVFile() {
+        Intent intent = new Intent(Intent.ACTION_GET_CONTENT);
+        intent.addCategory(Intent.CATEGORY_OPENABLE);
+        intent.setType("text/csv");
+        startActivityForResult(Intent.createChooser(intent, "Open CSV"), ACTIVITY_CHOOSE_FILE);
```
Comment (Karljoones) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/ProductListsActivity.java]: Please extract this string

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/ProductListsActivity.java [hunk 2]
```diff
@@ -145,4 +164,99 @@ public void onRightClicked(int position) {
...
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.menu_product_lists, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        int id = item.getItemId();
+        if (id == R.id.action_import_csv) {
```
Comment (Karljoones) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/ProductListsActivity.java]: Please use a switch statement here.

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -965,5 +965,8 @@
     <string name="settings_country_summary">Change your selected country</string>
     <string name="settings_country_title">Country</string>
     <string name="txt_no_nutrition_data">This product doesn\'t have any nutrition information on the packaging. If this has changed, you can edit the product.</string>
+    <string name="import_csv_to_list">Import CSV</string>
+    <string name="toast_import_csv_error">Error in importing csv file</string>
```
Comment (Karljoones) [inline app/src/main/res/values/strings.xml]: Please keep capitalisation consistent

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -965,5 +965,8 @@
     <string name="settings_country_summary">Change your selected country</string>
     <string name="settings_country_title">Country</string>
     <string name="txt_no_nutrition_data">This product doesn\'t have any nutrition information on the packaging. If this has changed, you can edit the product.</string>
+    <string name="import_csv_to_list">Import CSV</string>
```
Comment (Karljoones) [inline app/src/main/res/values/strings.xml]: Give more information here "Import Product CSV"

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/ProductListsActivity.java [hunk 5]
```diff
@@ -145,4 +164,99 @@ public void onRightClicked(int position) {
...
+
+    private void selectCSVFile() {
+        Intent intent = new Intent(Intent.ACTION_GET_CONTENT);
+        intent.addCategory(Intent.CATEGORY_OPENABLE);
+        intent.setType("text/csv");
+        startActivityForResult(Intent.createChooser(intent, "Open CSV"), ACTIVITY_CHOOSE_FILE);
+    }
+
+    @SuppressLint("StaticFieldLeak")
+    class parseCSV extends AsyncTask<Void, Integer, Boolean> {
```
Comment (Karljoones) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/ProductListsActivity.java]: It's good practice to use a Capital at the start of a class name. Please rename this to ParseCVS

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (teolemon) [inline app/src/main/res/values/strings.xml]:
```suggestion
    <string name="toast_import_csv_error">Error in importing CSV file</string>
```

File: app/src/main/res/values/strings.xml (hunk 4 above)
Comment (teolemon) [inline app/src/main/res/values/strings.xml]:
```suggestion
    <string name="import_csv_to_list">Import an Open Food Facts list CSV file</string>
```

**Comment by teolemon** [comment]
@prashantkh19 can you add a screenshot of the feature in the PR ?

**Comment by prashantkh19** [comment]
> @prashantkh19 can you add a screenshot of the feature in the PR ?

**Comment by teolemon** [comment]
Proposition:
image
Given that we're likely to add more options, I'd be tempted to "hide" the icon within an overflow button (which also feels more natural in this place)

**Comment by Karljoones** [comment]
Good to go once the conflicts are resolved

**Comment by fhqhwgads** [comment]
This issue seems to be breaking the build:
- It reverts the `androidx.*` imports to `android.support.*`
- It imports `com.opencsv.*` when opencsv was replaced by `org.apache.commons:commons-csv`

**Pull request merged**